src/Operation/IsNotEmpty.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace loophp\collection\Operation;
+
+use Closure;
+use Generator;
+use loophp\iterators\IterableIteratorAggregate;
+
+/**
+ * @immutable
+ *
+ * @template TKey
+ * @template T
+ */
+final class IsNotEmpty extends AbstractOperation
+{
+    /**
+     * @return Closure(iterable<TKey, T>): Generator<int, bool>
+     */
+    public function __invoke(): Closure
+    {
+        return static fn (iterable $iterable): Generator => yield (bool)(new IterableIteratorAggregate($iterable))->getIterator()->valid();
```

Comment:
Is the `(bool)` cast mandatory here ?
Is `->valid()` not returning a boolean already?

Reply:
Yes, you're right. Changed.